packages/lib/src/toolbar/controls/Selector/Selector.tsx

```diff
@@ -28,54 +37,136 @@ function Selector<T extends string>(props: Props<T>) {
     disabled,
     onChange,
     options,
-    optionComponent: Option,
+    optionComponent: OptionComp,
   } = props;
 
-  const { height: winHeight } = useWindowSize();
-  const menuMaxHeight = winHeight - MENU_TOP - MENU_BOTTOM;
+  const [isOpen, toggle] = useToggle();
+  const [activeIndex, setActiveIndex] = useState<number | null>(null);
+  const [selectedIndex, setSelectedIndex] = useState(() => {
+    return getAllOptions(options).indexOf(value);
+  });
+
+  const labelId = useId();
+  const referenceId = useId();
+  const selectedOptionId = useId();
+  const listRef = useRef<(HTMLButtonElement | null)[]>([]);
+
+  const { refs, floatingStyles, context } = useFloating<HTMLButtonElement>({
+    open: isOpen,
+    middleware: [
+      offset(6),
+      size({
+        padding: 12,
+        apply({ availableHeight, elements, rects }) {
+          Object.assign(elements.floating.style, {
+            maxHeight: `${Math.min(availableHeight, MENU_MAX_HEIGHT)}px`,
+            minWidth: `${rects.reference.width}px`,
+          });
+        },
+      }),
```

Comment:
This is an improved version of the `floatingMinWidth` middleware that also sets the max height of the popup so it never overflows the viewport. I will use it in `ExportEntry` as well eventually.